packages/wasm/src/transaction.ts

```diff
@@ -5,26 +5,28 @@ import {
   TransactionView,
 } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1/transaction_pb';
 import type { IdbConstants } from '@penumbra-zone/types/indexed-db';
-import { JsonValue } from '@bufbuild/protobuf';
 import { FullViewingKey } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb';
-
-interface TxInfoWasmResult {
-  txp: JsonValue;
-  txv: JsonValue;
-}
+//import { typeRegistry } from '@penumbra-zone/protobuf';
 
 export const generateTransactionInfo = async (
   fullViewingKey: FullViewingKey,
   tx: Transaction,
   idbConstants: IdbConstants,
 ) => {
-  const { txp, txv } = (await transaction_info(
+  const txInfo: unknown = await transaction_info(
     fullViewingKey.toBinary(),
     tx.toBinary(),
     idbConstants,
-  )) as TxInfoWasmResult;
+  );
+
+  if (!Array.isArray(txInfo) || !Array.isArray(txInfo[0]) || !Array.isArray(txInfo[1]))
+    throw new Error('Invalid transaction info');
```

Comment:
Can you convert this to a type guard? Something like:

```ts
type TransactionInfo = [number[], number[]];

const isTransactionInfo = (data: unknown): data is TransactionInfo =>
  Array.isArray(data) &&
  data.length >= 2 &&
  Array.isArray(data[0]) &&
  data[0].every(item => typeof item === 'number') &&
  Array.isArray(data[1]) &&
  data[1].every(item => typeof item === 'number');
```

and then you can do:
```ts
  if (!isTransactionInfo(txInfo)) throw new Error('Invalid transaction info');

  const txpBin = new Uint8Array(txInfo[0]);
  const txvBin = new Uint8Array(txInfo[1]);
```

Reply:
deserialization will throw if they're malformed. also, the arrays may be a few thousand items long, so mapping is excessive.

Reply:
the correct thing would be to ensure wasm is returning TypedArrays, i'll spend some time on that in the morning